web/src/components/Navigation/TopNav.tsx

```diff
@@ -116,42 +123,44 @@ export default function TopNav({ searchValue, setSearchValue }: Props) {
                 className={"w-11/12 max-w-xl sm:w-2/3"}
               />
               <LinkIconButton
-                href={"/create"}
+                href={`/create?prompt=${searchValue}`}
                 className={"hidden text-base sm:block"}
               >
                 <MdAddCircle className={"h-full w-10"} />
               </LinkIconButton>
             </motion.li>
           )}
 
-          {isProfile ? (
+          {(!isProfile || !isOwnProfile) && (
             <motion.li
-              key={"settings"}
+              key={"profile"}
               variants={positionVariants}
               initial={"initialRight"}
               animate={"animate"}
               exit={"initialRight"}
-              whileHover={{ rotate: 20 }}
               transition={transitions.springStiff}
-              className={"ml-auto"}
+              className={"hidden sm:block"}
             >
-              <IconButton>
-                <MdSettings className={"h-full w-10"} />
-              </IconButton>
+              <LinkIconButton href={`/profile/${userName}`}>
```

Comment:
Does next uri encode these for us? If not, probably should in case someone picks a weird username.

Reply:
I can't find a definitive answer on this but I think nextjs internally encodes these strings

Reply:
We can make usernames only be alphanumerical characters

Reply:
That can work, although probably wouldn't hurt to call the escaping function just to be safe